Domains picker: update copy to reflect monthly vs annual plans difference

ciampo opened the pull request:
## Changes proposed in this Pull Request

* Update copy in domain picker package, showing that custom domains are included with _annual_ plans
* The same copy is now shared across all flows (gutenboarding, step by step launch, focused launch)
* Fix viewport `isMobile` check, so that there isn't a 1px overlap between CSS and JS

Do not merge this PR until all translations have been completed!

## Testing instructions

### Setup
- prepare your sandbox
- in two separate terminal windows, run:
  - `yarn start` 
  - `cd apps/editing-toolkit && yarn dev --sync`

### Focused Launch

1. Assign yourself to the "treatment" group in the `focused_launch_flow_v2` A/B test (id `20106`)
2. Add an _unlaunched_ site created through `/start`to your sandbox
3. Open the block editor (e.g. by visiting `calyspo.localhost:3000/page/UNLAUNCHED_SITE_CREATED_WITH_START.wordpress.com/home`)`
4. Press "Launch" button, Focused Launch summary view should appear
    - [x] Verify that the new copy is correct in the domain picker step (both mobile and desktop), according to A/C in #49256
    - [x] These copy changes are not expected to affect non-english locales for the purpose of this PR (but it's important that the previous _translated_ fallback text is used)

### Step by Step Launch

1. Add an _unlaunched_ site on a _free_ plan created through `/new` to your sandbox
2. Visit `wordpress.com/page/SITE_ID.wordpress.com/home`
3. Press "Launch" button, Step by Step Launch modal should appear. Navigate to domain step
    - [x] Verify that the new copy is correct in the domain picker step (both mobile and desktop), according to A/C in #49256
    - [x] These copy changes are not expected to affect non-english locales for the purpose of this PR (but it's important that the previous _translated_ fallback text is used)

### Gutenboarding

1. Visit `/new` and navigate to the domain picker
2. Verify that the new copy is correct in the domain picker step (both mobile and desktop), according to A/C in #49256

## Screenshots

Context for translators:

<img width="602" alt="focused-launch-mobile-big-new" src="https://user-images.githubusercontent.com/1083581/105831373-e9773880-5fc6-11eb-8cbb-cf445f1d0c95.png">


### Visual comparison of Gutenboarding changes

<details>
  <summary>Click to expand</summary>

  |     | Before | After |
  |---|---|---|
  | Mobile | <img width="448" alt="gutenboarding-mobile-old" src="https://user-images.githubusercontent.com/1083581/105831785-715d4280-5fc7-11eb-9ad3-5789462be107.png"> | <img width="449" alt="gutenboarding-mobile-new" src="https://user-images.githubusercontent.com/1083581/105831806-76ba8d00-5fc7-11eb-8149-e7724c6ae090.png"> | 
  | Mobile big |  <img width="610" alt="gutenboarding-mobile-big-old" src="https://user-images.githubusercontent.com/1083581/105831838-82a64f00-5fc7-11eb-8d11-6e99a58b2676.png"> | <img width="610" alt="gutenboarding-mobile-big-new" src="https://user-images.githubusercontent.com/1083581/107504529-bbcfe900-6b9b-11eb-9b0c-a986b4e60104.png"> | 
  | Desktop | <img width="1242" alt="gutenboarding-desktop-old" src="https://user-images.githubusercontent.com/1083581/105832091-d022bc00-5fc7-11eb-90ea-5191c12304e4.png"> | <img width="1242" alt="gutenboarding-desktop-new" src="https://user-images.githubusercontent.com/1083581/107504423-8f1bd180-6b9b-11eb-8e60-cc0a637a3de0.png"> | 
</details>


### Visual comparison of Focused Launch changes

<details>
  <summary>Click to expand</summary>
  
  |     | Before | After |
  |---|---|---|
  | Mobile | <img width="361" alt="focused-launch-mobile-old" src="https://user-images.githubusercontent.com/1083581/105832085-cf8a2580-5fc7-11eb-88b2-8d6e3f2d85dd.png"> | <img width="361" alt="focused-launch-mobile-new" src="https://user-images.githubusercontent.com/1083581/105832084-cef18f00-5fc7-11eb-86bf-5956266c9992.png"> | 
  | Mobile big | <img width="601" alt="focused-launch-mobile-big-old" src="https://user-images.githubusercontent.com/1083581/105832082-ce58f880-5fc7-11eb-9213-050ef0d0b9a9.png"> | <img width="602" alt="focused-launch-mobile-big-new" src="https://user-images.githubusercontent.com/1083581/105832080-cdc06200-5fc7-11eb-81a4-25e1e19b29fc.png"> | 
  | Desktop | <img width="1242" alt="focused-launch-desktop-old" src="https://user-images.githubusercontent.com/1083581/105832077-cc8f3500-5fc7-11eb-8e27-a1b5254fae65.png"> | <img width="1239" alt="focused-launch-desktop-new" src="https://user-images.githubusercontent.com/1083581/105832072-ca2cdb00-5fc7-11eb-881e-bb64b50a95eb.png"> | 
</details>


### Visual comparison of Step by Step launch changes

<details>
  <summary>Click to expand</summary>
  
  |     | Before | After|
  |---|---|---|
  | Mobile | <img width="361" alt="step-by-step-mobile-old" src="https://user-images.githubusercontent.com/1083581/105832106-d31dac80-5fc7-11eb-886d-e7215dd83f38.png"> | <img width="360" alt="step-by-step-mobile-new" src="https://user-images.githubusercontent.com/1083581/105832101-d2851600-5fc7-11eb-9b56-0c97a2fd373d.png"> | 
  | Mobile big | <img width="602" alt="step-by-step-mobile-big-old" src="https://user-images.githubusercontent.com/1083581/105832098-d2851600-5fc7-11eb-9e73-212aea549393.png"> | <img width="601" alt="step-by-step-mobile-big-new" src="https://user-images.githubusercontent.com/1083581/105832096-d1ec7f80-5fc7-11eb-84b6-24a7716ac90f.png"> | 
  | Desktop | <img width="1241" alt="step-by-step-desktop-old" src="https://user-images.githubusercontent.com/1083581/105832095-d153e900-5fc7-11eb-869e-d642d3fb79e2.png"> | <img width="1241" alt="step-by-step-desktop-new" src="https://user-images.githubusercontent.com/1083581/105832092-d0bb5280-5fc7-11eb-8d0e-6ba3b360bf2a.png"> | 
</details>

Fixes #49256
Relates to #47576

ciampo committed:
New copy across all flows

ciampo committed:
Fix viewport check, so that there isn't a 1px overlap between CSS and JS

a8ci18n commented on packages/domain-picker/src/domain-picker/suggestion-item.tsx:
```diff
@@ -106,20 +106,15 @@ const DomainPickerSuggestionItem: React.FC< Props > = ( {
-	const firstYearIncludedInPaid = isMobile
-		? __( 'Included in paid plans', __i18n_text_domain__ )
+	const paidIncludedDomainLabel = isMobile
+		? __( 'Included in annual plans', __i18n_text_domain__ )
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Included with annual plans' )` *ES Score: 10*

a8ci18n commented on packages/domain-picker/src/domain-picker/suggestion-item.tsx:
```diff
@@ -106,20 +106,15 @@ const DomainPickerSuggestionItem: React.FC< Props > = ( {
+		? __( 'Included in annual plans', __i18n_text_domain__ )
 		: createInterpolateElement(
-				__( '<strong>First year included</strong> in paid plans', __i18n_text_domain__ ),
+				__( '<strong>First year included</strong> in annual plans', __i18n_text_domain__ ),
```
:warning: This change will be queued for retranslation. We'll display the English text until then.

ciampo commented on packages/domain-picker/src/domain-picker/suggestion-item.tsx:
```diff
@@ -92,7 +92,7 @@ const DomainPickerSuggestionItem: React.FC< Props > = ( {
 	const localizeUrl = useLocalizeUrl();
 
-	const isMobile = useViewportMatch( 'small', '<' );
+	const isMobile = ! useViewportMatch( 'small', '>=' );
```
This prevents a small bug where, when the viewport is 600px wide, this check would previously return `true`, while the CSS for mobile styles would only be applied below 599px

ciampo commented:
Rebased on top of latest changes

ciampo committed:
Rename variables

StefanNieuwenhuis commented:
#### Gutenboarding testing instructions

Visited URL: `/new/domains-modal`

> In the Domain step, on a mobile viewport, the green copy next to each domain item should say "Included in annual plans" (https://github.com/Automattic/wp-calypso/issues/49256)

There is no such copy on mobile viewports.

<img width="386" alt="Screenshot 2021-02-09 at 16 35 47" src="https://user-images.githubusercontent.com/12104589/107387422-16fad080-6af5-11eb-8f12-4a69de929ac5.png">

#### Step-by-step-launch testing instructions

> In the Domain step, on a mobile viewport, the green copy next to each domain item should say "Included in annual plans"(https://github.com/Automattic/wp-calypso/issues/49256)

There is no such copy on mobile viewports.

StefanNieuwenhuis reviewed (commented):
Code LGTM but can you please address https://github.com/Automattic/wp-calypso/pull/49290#issuecomment-776030080?

ciampo commented:
> Code LGTM but can you please address [#49290 (comment)](https://github.com/Automattic/wp-calypso/pull/49290#issuecomment-776030080)?

Nice catch! In fact, the A/C were wrong for Gutenboarding and Step by Step flows — I just updated them.

As a further confirmation:
- You can look in the PR description, I had already added screenshots of the expected result for each flow / viewport size
- You can see that normally that text is [hidden on mobile viewports](https://github.com/Automattic/wp-calypso/blob/trunk/packages/domain-picker/src/domain-picker/style.scss#L396), and is [shown only for the "individual item" variation](https://github.com/Automattic/wp-calypso/blob/trunk/packages/domain-picker/src/domain-picker/style.scss#L408-L410)

StefanNieuwenhuis commented:
#### Gutenboarding 

As discussed, can you please make the "First year included" part of the copy bold?

<img width="338" alt="Screenshot 2021-02-10 at 10 08 35" src="https://user-images.githubusercontent.com/12104589/107488661-f596f480-6b87-11eb-97c1-eaca0a397c26.png">

ciampo committed:
Domain picker: force <strong> text to be be bold

ciampo commented:
> #### Gutenboarding
> As discussed, can you please make the "First year included" part of the copy bold?

Hey @StefanNieuwenhuis , I've addressed this change in code and update the screenshots in the PR description

StefanNieuwenhuis reviewed (approved):
LGTM! 🚢

ciampo commented:
Waiting for strings to be translated

alshakero reviewed (approved):
Works as expected. But wanted to bring it to your attention that in step-by-step. The green message is "Included with annual plans". Not "First year included in annual plans".

ciampo commented:
@Automattic/i18n There seems to be an issue with the translations on this PR — Everything should already be translated, but I did one last check before merging and found out that the copy is still appearing in english

![image](https://user-images.githubusercontent.com/1083581/109473764-098d9200-7a74-11eb-9137-06fa46490908.png)

Can you help debugging what's the issue?

ciampo committed:
Use old translations as fallback until new translations are in

ciampo commented:
@StefanNieuwenhuis and @alshakero , I pushed some changes so that the UI will display the old text until the new text is fully translated. This is necessary for ETK related changes.

I updated the testing instructions reflecting the latest changes

yansern reviewed (approved):
Tested on Gutenboarding, Step-By-Step Launch & Focused Launch on both desktop & mobile view. LGTM.

ciampo commented on packages/domain-picker/src/domain-picker/suggestion-item.tsx:
```diff
@@ -76,42 +78,40 @@ const DomainPickerSuggestionItem: React.FC< Props > = ( {
+	);
+	const includedLabelFormatted =
+		locale === 'en' ||
+		hasTranslation?.( '<strong>First year included</strong> in annual plans', __i18n_text_domain__ )
```
Hey @Automattic/i18n , I've noticed that this check seems to return `false` in Gutenboarding even if the [strings should be available](https://translate.wordpress.com/projects/wpcom/-all-translated/632216/) — do you know why this is happening?

As a comparison, a similar check [a few lines above (line 90)](https://github.com/Automattic/wp-calypso/pull/49290/files#diff-df652cccef87aecec53f79cf4fd560f677b5af1c74439cfb7a7bbbff2da584c3R90) seems to be working fine in my tests.

As an additional check, on my local machine I tried removing the second argument to this call — and `hasTranslation?.( '<strong>First year included</strong> in annual plans' )` seems to work.

Any clues on why this is happening?

yansern replied:
Just pasting what I'm seeing when trying to debug `hasTranslation@packages/react-i18n/src/index.tsx`.

```
function hasTranslation(localeData, singular, context) {
  const key = typeof context === 'string' ? ''.concat(context, CONTEXT_DELIMETER, singular) : singular;
  return key in localeData;
}
```

![image](https://user-images.githubusercontent.com/1287077/109691221-83627000-7b87-11eb-80d5-aec391c70027.png)

yansern replied:
Perhaps we can just check for the translation without the context, e.g.
```
hasTranslation?.( '<strong>First year included</strong> in annual plans' )
```

That kinda fits the logic written in `hasTranslation`.

yuliyan replied:
The problem is that you're passing `__i18n_text_domain__` as a second parameter to `hasTranslation` where it expects `context`, and those two are different things. Given that translations in Calypso are stored in a single domain (applies for `i18n-calypso` and `@automattic/react-i18n`), I think it's fine for now to use `hasTranslation` with just the string as only parameter.

ciampo replied:
Thank you @yuliyan — I confused `context` and `domain` 😅 I will remove the second parameter to the `hasTranslation` calls

ciampo replied:
Done in https://github.com/Automattic/wp-calypso/pull/49290/commits/bc4e2c9ef8b27c07ca7425448f9f950687dff1d9

ciampo committed:
Remove 2nd argument from hasTranslation call
